families/identity/Dockerfile

```diff
@@ -11,39 +11,38 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+# ------------------------------------------------------------------------------
 
 # docker build -f families/identity/Dockerfile -t sawtooth-identity-tp-local .
 
-# -------------=== signing build ===-------------
-
-FROM ubuntu:bionic
+FROM ubuntu:xenial
 
 RUN apt-get update \
- && apt-get install gnupg -y
-
-RUN echo "deb http://repo.sawtooth.me/ubuntu/ci bionic universe" >> /etc/apt/sources.list \
- && (apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 8AA7AF1F1091A5FD \
- || apt-key adv --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys 8AA7AF1F1091A5FD) \
- && apt-get update
-
-RUN apt-get install -y -q \
-    git \
-    python3 \
-    python3-stdeb
-
-RUN apt-get install -y -q \
-    python3-grpcio \
-    python3-grpcio-tools \
-    python3-protobuf
-
-RUN apt-get install -y -q \
-    python3-colorlog \
-    python3-toml \
-    python3-yaml \
-    python3-zmq
-
-RUN mkdir -p /var/log/sawtooth
-
-ENV PATH=$PATH:/project/sawtooth-core/bin
+ && apt-get install -y \
+ curl \
+ gcc \
+ libssl-dev \
+ libzmq3-dev \
+ pkg-config \
+ unzip
+
+# For Building Protobufs
+RUN curl -OLsS https://github.com/google/protobuf/releases/download/v3.5.1/protoc-3.5.1-linux-x86_64.zip \
+ && unzip protoc-3.5.1-linux-x86_64.zip -d protoc3 \
+ && rm protoc-3.5.1-linux-x86_64.zip
+
+RUN curl https://sh.rustup.rs -sSf > /usr/bin/rustup-init \
+ && chmod +x /usr/bin/rustup-init \
+ && rustup-init -y
+
+ENV PATH=$PATH:/project/sawtooth-core/bin:/protoc3/bin:/root/.cargo/bin \
+    CARGO_INCREMENTAL=0
 
 WORKDIR /project/sawtooth-core
+
+CMD cd families/identity/sawtooth_identity \
+ && echo "\033[0;32m--- Building identity-tp ---\n\033[0m" \
+ && rm -rf ./bin/ \
+ && mkdir -p ./bin/ \
+ && cargo build --release \
+ && cp ./target/release/sawtooth_identity ./bin/sawtooth_identity
```

Comment:
I think this is busted and should say `cp ./target/release/identity-tp ./bin/identity-tp` and that there are some additional places where this needs to be changed, such as `bin/identity-tp`